crates/oxc_linter/src/rules/import/no_amd.rs

```diff
@@ -0,0 +1,93 @@
+use oxc_ast::ast::Expression;
+use oxc_diagnostics::{
+  miette::{self, Diagnostic},
+  thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, ModuleKind, Span};
+use oxc_ast::AstKind;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-import(no-amd): Do not use AMD `require` and `define` calls.")]
+#[diagnostic(severity(warning), help("Consider change {1} to EMScript import or export"))]
+struct NoAmdDiagnostic(#[label] pub Span, Atom);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoAmd;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Forbid AMD `require` and `define` calls.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// // fail
+    /// require([a, b], function() {} );
+    /// // pass
+    /// require('../name');
+    /// require(`../name`);
+    /// ```
+    NoAmd,
+    nursery
+);
+
+/// https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-amd.js
+impl Rule for NoAmd {
+  fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+
+    let _ = match ctx.source_type().module_kind() {
+      ModuleKind::Module => (),
+      _ => return,
+    };
+
+    let call_expr = match node.kind() {
+      AstKind::CallExpression(call_expr) => call_expr,
+      _ => return,
+    };
+
+    let identifier = match &call_expr.callee {
+      Expression::Identifier(ref xx) => xx,
+      _ => return,
+    };
+
+    if identifier.name != "define" && identifier.name != "require" {
+      return;
+    }
+
+    if call_expr.arguments.len() != 2 { return; }
+    ctx.diagnostic(NoAmdDiagnostic(identifier.span, identifier.name.clone()));
+  }
+}
+
+#[test]
+fn test() {
+  use crate::tester::Tester;
+
+  let mut tester = Tester::new_without_config::<String>(NoAmd::NAME, vec![], vec![])
+    .with_import_plugin(true);
+
+  {
+    let pass = vec![
+      "var _ = require('lodash')",
+      "var find = require('lodash.find')",
+      "var foo = require('./foo')",
+      "var foo = require('../foo')",
+      "var foo = require('foo')",
+      "var foo = require('./')",
+      "var foo = require('@scope/foo')",
+      "var bar = require('./bar/index')",
+    ];
+
+    let fail = vec![
+      "require([a, b], function() {})",
+      "define([a, b], function() {})",
+    ];
+
+    tester = tester.change_rule_path("no-amd.js").update_expect_pass_fail(pass, fail);
+    tester.test_and_snapshot();
+  }
```

Comment:
just do like this?
```rs
    let pass = vec![
      "var _ = require('lodash')",
      "var find = require('lodash.find')",
      "var foo = require('./foo')",
      "var foo = require('../foo')",
      "var foo = require('foo')",
      "var foo = require('./')",
      "var foo = require('@scope/foo')",
      "var bar = require('./bar/index')",
    ];

    let fail = vec![
      "require([a, b], function() {})",
      "define([a, b], function() {})",
    ];

Tester::new_without_config::<String>(NoAmd::NAME, pass, fail)
    .with_import_plugin(true).test_and_snapshot

Reply:
Thanks, I updated the codes like you suggested.